src/main/java/edu/kit/datamanager/pit/pidsystem/impl/local/PidDatabaseObject.java

```diff
@@ -32,7 +32,22 @@ public class PidDatabaseObject {
     @Column(name = "pid")
     private String pid;
 
+    /**
+     * About the column definition we use here:
+     * 
+     * int javax.persistence.Column.length() (Optional) The column length. (Applies
+     * only if a string-valued column is used.) Default: 255
+     * 
+     * The h2 database we use for testing has a CHARACTER VARYING limit of
+     * 1_000_000_000
+     * (http://h2database.com/html/datatypes.html#character_varying_type).
+     * 
+     * Postgres does not have a limit according to
+     * https://www.postgresql.org/docs/current/datatype-character.html, but we
+     * assume it will not use its text datatype for the collection anyway.
+     */
     @ElementCollection(fetch = FetchType.EAGER)
+    @Column(length = 1_000_000_000)
```

Comment:
This line fixed the issue. Affected were only databases which map the collection to a String (character varying) anyway, so for example it should not affect Postgres at all (because it does not use a string and the length only affects strings in this column).

All other changes are tests or comments.

Reply:
Sure, this should help, but seems like an overkill. ;-) I'm not sure how affected databases deal with such numbers, in the worst case, a lot of memory (> 1 GB) might be reserved for situations, where many entries are in this column. However, such databases should never be used in production anyways, so something smaller could be sufficient including a short note in the configuration file, that there are such limitations.

Something which concerns me more is this one: 

[https://dev.mysql.com/doc/refman/8.0/en/char.html#:~:text=Values%20in%20VARCHAR%20columns%20are,value%20from%200%20to%2065%2C535.](https://dev.mysql.com/doc/refman/8.0/en/char.html#:~:text=Values%20in%20VARCHAR%20columns%20are,value%20from%200%20to%2065%2C535.)

According to that, there might be a limitation of varchar columns for some databases (which are actually not affected) and I'm not sure how they deal with such numbers.

Reply:
- My test needs at least a length of ~6500 (100 attributes with 100 values each, each attribute or value being a PID with maybe a length of ~20).
- > [For MySQL] The length can be specified as a value from 0 to 65,535. (https://dev.mysql.com/doc/refman/8.0/en/char.html) (I do not believe there would be such a ridiculous mapping being done in MySQL)
- > [For SQLite] The default value of this macro is 1 billion (1 thousand million or 1,000,000,000). (https://www.sqlite.org/limits.html#max_length) (Not sure if it would be affected from the weird mapping, but the default is already high here, so actually we would restrict it, if it would be affected)
- Same for h2, but default was 255

My impression is that databases with limited capabilities / data types like h2 tend to have a higher varchar limit for compensation. I'd suggest either 10_000 (6500 + some buffer) or the upper limit of MySQL if the goal is to not make it too large. What do you think?

Reply:
I assume the limits of MySQL to be fine and sufficient. As soon as we are talking about records of this size, we might have performance issues anyways so let's avoid them by technical limitation. ;-)